PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,468 @@
+        It 'matches a little or matches a lot' {
+            <#
+            A number of the quantifiers have two versions: A greedy version which tries to match an
+            element as many times as possible, and a non-greedy (or lazy) version which tries to
+            match an element as few times as possible. You can turn a greedy quantifier into a lazy
+            quantifier by simply adding a ?.
+            #>
+
+            $greedy = 'o+'
+            $lazy = 'o+?'
+            '____' | Should -Be @([regex]::Matches('helloooooo', $greedy).Value)[0]
+            '____' | Should -Be @([regex]::Matches('helloooooo', $lazy).Value)[0]
+        }
+    }
+}
+
+Describe 'Special Symbols' {
+    <#
+        In regex, there are far more symbols with unique meanings than just quantifiers. What comes
+        next is not a definitive guide to EVERY SINGLE ONE, but rather an introduction to some of
+        the most common and useful special symbols in regex.
+        
+        This is ^\where things st\Art getting \weir\D$
+    #>
+    
+
+    Context '. (period)' {
+        It 'matches any character' {
+            # The . (period) matches literally any character
+    
+            [bool]$morningMatch = $____ # Should be either $true or $false
+            $morningMatch | Should -Be 'Lazy Sunday mornings' -match 'S.nday'
+    
+            '____' -match 'invi.a.ion' | Should -BeTrue
+        }
+    }
+
+    Context '\n' {
+        It 'matches new lines' {
+            <#
+                The \n symbol is the first regex symbol you've encountered that isn't just a single
+                character. Normally if you saw a lowercase "n" in a regex, it would just mean
+                "literally the letter n". When it's preceded by a backslash, however, that "n" takes
+                on a special meaning. In this case, \n matches new lines.
+    
+                There are many more regex symbols that are single letters preceded by a backslash.
+                In fact, the backslash is probably the single most important character in all of
+                regex.
+            #>
+    
+            $multiLine = @"
+They might look
+similar and look
+soft, but you should
+never confuse a
+wild cougar for
+a domestic cat
+"@
+            $catPattern = '____\n'
+            $multiLine -match $catPattern | Should -BeTrue
+        }
+    }
+
+    Context '\d and \D' {
+        It 'matches digits and non-digits' {
+            <#
+                The \d symbol matches digits (the numbers 0 - 9 inclusive), while the symbol \D
+                matches anything OTHER THAN a digit. Yes, regular expressions are case sensitive.
+                This "lower case for affirmative, upper case for negative" pattern is common among
+                regex symbols as you'll discover as you carry on.
+            #>
+    
+            '__' -match '\d' | Should -BeTrue
+            '__' -match '\d\d\d' | Should -BeTrue
+            '__' -match '\D' | Should -BeTrue
```

Comment:
Separate it blocks! Probably two is plenty, one fo the positive and one for the negatives:
`It 'matches digits with \d'` and `It 'matches non-digits with \D'`